components/Dialogs/src/MDCAlertController.h

```diff
@@ -69,6 +70,11 @@
  */
 @property(nonatomic, nonnull, readonly) NSArray<MDCAlertAction *> *actions;
 
+/**
+ The MDCAlertControllerView instance that is being presented by this view controller.
+ */
+@property(nonatomic, nonnull, readonly) MDCAlertControllerView *alertControllerView;
```

Comment:
I'm a bit concerned about the implications of exposing the entire view - did you consider exposing only the font/color knobs here and forwarding them to the view under the hood? This would reduce our public API contract in case we move away from this `MDCAlertControllerView` implementation in the future.

Reply:
MDCAlertControllerView is the same as controller.view which is already exposed though. so I don't think this could add any extra implications.

Reply:
I agree that that is the case right now, but `MDCAlertControllerView` is an implementation detail that might change in the future. The `MDCAlertControllerView` contract may grow over time, or fonts/colors may need to be set on a different view.

If we use the controller as a layer of abstraction then we are able to hide this underlying implementation detail from clients, giving us more long-term flexibility to improve upon the API.

Another benefit of this approach is that we don't have to call `[self loadViewIfNeeded]`, because we'll simply map the properties to the underlying views when set or upon view creation.

Reply:
We are already exposing the view for UI_Appearance, so it is not really an implementation detail. It actually looks pretty confusing being able to set the values via UI_Appearance to MDCAlertControllerView, while MDCAlertController does not even have any documentation on having a MDCAlertControllerView. We had a similar approach on FeatureHighlight which as well looked pretty confusing IMO, I had a discussion with Ian and we decided to take this approach.
I agree that a layer of abstraction could have some benefits, but we really don't have that abstraction right now anyway.

Reply:
Sorry I am still confused, exposing the view APIs is the same as view controller's view.
any other property on MDCAlertControllerView is what we are currently exposing which is the same what we are exposing for UIAppearance proxy.

Before this PR, if we decide to change MDCAlertControllerView to some other implementation, we have to clear all dependencies for setting color and font to MDCAlertControllerView and switch it to potential implementation.
After this PR same amount of work would be required. I really think think exposing the view would be a safe bet here and cleaner approach since we are removing one extra layer of communication.

Reply:
@ianegordon @jverkoey

Reply:
I feel that you want a client to know that the view is an instance of `MDCAlertControllerView`, but I can't quite understand why.

Can you help me see why it's important that `MDCAlertControllerView` be exposed to a client as the view's type?

If the primary motivation is "removing one extra layer of communication", I want to emphasize that this layer of indirection is intentionally desired in order to give us the flexibility in the future to modify the underlying implementation with less concern about breaking clients.

Reply:
I made  the change and removed MDCAlertControllerView from the public API.